docs/integration-guide.md

```diff
@@ -107,6 +107,342 @@ Here's how we recommend doing it:
+
+#### Step 1: Create main webhook code
+
+The majority of the code for your webhook integration will be in a single
+python file in `zerver/views/webhooks/`.
+
+The Hello World integration is in `zerver/views/webhooks/helloworld.py`:
+
+```
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.lib.validator import check_dict, check_string
+
+@api_key_only_webhook_view('HelloWorld')
+@has_request_variables
+def api_helloworld_webhook(request, user_profile, client,
+                           payload=REQ(argument_type='body'),
+                           stream=REQ(default='test'),
+                           topic=REQ(default='Hello World')):
+
+  # construct the body of the message
+  body = 'Hello! I am happy to be here! :smile: '
+
+  # try to add the Wikipedia article of the day
+  # return appropriate error if not successful
+  try:
+      body_template = '\nThe Wikipedia featured article for today is **[{featured_title}]({featured_url})**'
+      body += body_template.format(**payload)
+  except KeyError as e:
+      return json_error(_("Missing key {} in JSON").format(e.message))
+
+  # send the message
+  check_send_message(user_profile, client, 'stream', [stream], topic, body)
+
+  # return json result
+  return json_success()
+
+```
+
+The above code imports the required functions and defines the main webhook
+function `api_helloworld_webook`, decorating it with `api_key_only_webhook_view` and
+`has_request_variables`.
+
+You must pass the name of your webhook to the `api_key_only_webhook_view`
+decorator. Here we have used `HelloWorld`.
+
+You may name your webhook function whatever you like, though it's a good idea
+to be consistent with other webhook integrations. We recommend following the
+format `api_webhookname_webhook`.
+
+At minimum, the webhook function must accept `request`, `user_profile`, and
+`client`. You may also want to define additional parameters using the `REQ`
+object.
+
+In the example above, we have defined `payload` which is populated from the
+body of the http request, `stream` with a default of `test` (available by
+default in Zulip dev environment), and `topic` with a default of `Hello World`.
+
+In the body of the function we define the body of the message as `Hello! I am
+happy to be here! :smile: `. The `:smile:` indicates an emoji. Then we append a
+link to the Wikipedia article of the day as provided by the json payload. If
+the json payload does not include data for `featured_title` and `featured_url`
+we catch a `KeyError` and use `json_error` to return the appropriate
+information: a 400 http status code with relevant details.
+
+Then we send a public (stream) message with `check_send_message`.  Finally, we
```

Comment:
maybe write as "`check_send_message`, which will validate the message and then send it."